src/inspector/inspector.service.ts

```diff
@@ -3,60 +3,51 @@ import { Inject, Injectable, LoggerService, OnModuleInit } from '@nestjs/common'
+  public latestProcessedEpoch = 0n;
+
   public constructor(
     @Inject(LOGGER_PROVIDER) protected readonly logger: LoggerService,
     protected readonly config: ConfigService,
     protected readonly clClient: ConsensusProviderService,
     protected readonly storage: ClickhouseService,
-    protected readonly dataProcessor: DataProcessingService,
-    protected readonly statsProcessor: StatsProcessingService,
-    protected readonly criticalAlertService: CriticalAlertsService,
-    protected readonly slotsCacheService: BlockCacheService,
+    protected readonly prometheus: PrometheusService,
+    protected readonly criticalAlerts: CriticalAlertsService,
+
+    protected readonly dutyService: DutyService,
+    protected readonly dutyMetrics: DutyMetrics,
   ) {}
 
   public async onModuleInit(): Promise<void> {
     this.logger.log(`Starting slot [${this.config.get('START_SLOT')}]`);
+    this.latestProcessedEpoch = await this.storage.getMaxEpoch();
+    this.prometheus.epochTime = await this.clClient.getSlotTime(this.latestProcessedEpoch * 32n);
+    this.prometheus.epochNumber.set(Number(this.latestProcessedEpoch));
   }
 
   public async startLoop(): Promise<never> {
-    this.logger.log(`DRY RUN ${this.config.get('DRY_RUN') ? 'enabled' : 'disabled'}`);
-
     const version = await this.clClient.getVersion();
     this.logger.log(`Beacon chain API info [${version}]`);
-    this.logger.log(
-      `Calculated slot step [${this.config.get('FETCH_INTERVAL_SLOTS')}] based on $FETCH_INTERVAL_SECONDS / $CHAIN_SLOT_TIME_SECONDS`,
-    );
 
     // eslint-disable-next-line no-constant-condition
     while (true) {
       try {
-        // Calculate finalized data stats (validator balances, attestations, proposes and etc.)
-        const nextFinalizedSlot = this.calculateNextFinalizedSlot();
-        this.slotsCacheService.purgeOld(nextFinalizedSlot / BigInt(this.config.get('FETCH_INTERVAL_SLOTS')));
-        const { slotToWrite, stateRoot, slotNumber } = await this.waitForNextFinalizedSlot(nextFinalizedSlot);
-        if (slotToWrite > 0) {
-          const res = await this.dataProcessor.processAndWriteFinalizedData(slotToWrite, stateRoot, slotNumber);
-          let possibleHighRewardValidators: string[] = [];
-          if (res?.userIDs.length) {
-            const headEpoch = await this.calculateHeadEpoch();
-            possibleHighRewardValidators = await this.dataProcessor.getPossibleHighRewardValidatorIndexes(res.userIDs, headEpoch);
-          }
-          await this.statsProcessor.calculateUserStats(slotToWrite, possibleHighRewardValidators);
-          await this.statsProcessor.calculateOtherStats(res.otherValidatorsCounts, res.otherAvgSyncPercent);
-          await this.statsProcessor.finalizeAppIterate(slotToWrite);
-          await this.criticalAlertService.sendCriticalAlerts(slotToWrite);
+        const { epoch, stateSlot } = await this.waitForNextFinalizedSlot();
+        if (epoch > 0) {
```

Comment:
What's the case?

Reply:
Log about what?

Reply:
The special case `epoch <= 0` caused by timeout.

Reply:
https://github.com/lidofinance/ethereum-validators-monitoring/blob/a1b64f20f6a5bdf02c98b106319b44bd737706d9/src/inspector/inspector.service.ts#L72

Reply:
https://github.com/lidofinance/ethereum-validators-monitoring/blob/a1b64f20f6a5bdf02c98b106319b44bd737706d9/src/inspector/inspector.service.ts#L75-L77